src/core/tsi/transport_security.h

```diff
@@ -111,6 +111,9 @@ tsi_result tsi_construct_allocated_string_peer_property(
 tsi_result tsi_construct_string_peer_property_from_cstring(
     const char *name, const char *value, tsi_peer_property *property);
 
+/* Destroy plugin registration funcion. */
+void tsi_register_destroy_plugin(void (*destroy)(void));
```

Comment:
Design question: Why are we creating a new registration system for TSI implementations, rather than using the existing gRPC plugin system?  In other words, can't we just have each TSI implementation be its own gRPC plugin, in which case there's no need for this duplicate machinery?